src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/CompilerServices/NewLateBinding.vb

```diff
@@ -0,0 +1,1503 @@
+            Dim lastArgument As Object = Nothing
+
+            If HasFlag(lookupFlags, BindingFlagsSetProperty) Then
+                If arguments.Length = 0 Then
+                    failure = OverloadResolution.ResolutionFailure.InvalidArgument
+
+                    If reportErrors Then
+                        Throw New InvalidCastException(
+                            GetResourceString(SR.PropertySetMissingArgument1, methodName))
+                    End If
+
+                    Return Nothing
+                End If
+
+                savedArguments = arguments
+                arguments = New Object(argumentCount - 2) {}
+                System.Array.Copy(savedArguments, arguments, arguments.Length)
+                lastArgument = savedArguments(argumentCount - 1)
+            End If
+
+            Dim resolutionResult As Method =
+                ResolveOverloadedCall(
+                    methodName,
+                    members,
+                    arguments,
+                    argumentNames,
+                    typeArguments,
+                    lookupFlags,
+                    reportErrors,
+                    failure,
+                    baseReference)
+
+            Debug.Assert(failure = OverloadResolution.ResolutionFailure.None OrElse Not reportErrors,
+                         "if resolution failed, an exception should have been thrown")
+
+            If failure <> OverloadResolution.ResolutionFailure.None Then
+                Debug.Assert(resolutionResult Is Nothing, "resolution failed so should have no result")
+                Return Nothing
+            End If
+
+            Debug.Assert(resolutionResult IsNot Nothing, "resolution didn't fail, so should have result")
+
+#If BINDING_LOG Then
+            Console.WriteLine("== RESULT ==")
+            Console.WriteLine(ResolutionResult.DeclaringType.Name & "::" & ResolutionResult.ToString)
+            Console.WriteLine()
+#End If
+
+            'Overload resolution will potentially select one method before validating arguments.
+            'Validate those arguments now.
+            If Not resolutionResult.ArgumentsValidated Then
+
+                If Not CanMatchArguments(resolutionResult, arguments, argumentNames, typeArguments, False, Nothing) Then
+
+                    failure = OverloadResolution.ResolutionFailure.InvalidArgument
+
+                    If reportErrors Then
+                        Dim errorMessage As String = ""
+                        Dim errors As New List(Of String)
+
+                        Dim result As Boolean =
+                            CanMatchArguments(resolutionResult, arguments, argumentNames, typeArguments, False, errors)
+
+                        Debug.Assert(result = False AndAlso errors.Count > 0, "expected this candidate to fail")
+
+                        For Each errorString As String In errors
+                            errorMessage &= vbCrLf & "    " & errorString
```

Comment:
There are a handful of places in the source where vbCrLf is being used to introduce newlines... should vbNewLine be used instead, especially if it's changed to be Environment.NewLine and have the right newline characters cross-plat?